ruby-on-rails/docs/gem-tutorials/spree.md

```diff
@@ -0,0 +1,324 @@
+      redirect_to spree.account_url, notice: Spree.t(:account_updated)
+    else
+      prepare_error
+      redirect_to request.env['HTTP_REFERER']
+    end
+  end
+
+  private
+
+  def sanitize_params
+    return params[:user].permit(permitted_params)
+  end
+
+  def prepare_error
+    error = @user.errors.full_messages.join(', ')
+    flash[:error] = error
+  end
+
+  def permitted_params
+    %i[email login last_name first_name gender birthday]
+  end
+
+  def add_birthday
+    params[:user].merge(birthday: "#{params[:year]}-#{params[:month]}-#{params[:day]}")
+  end
+end
+```
+
+### Add a new controller
+
+This is a way to add new controller class in spree, nothing magical:
+
+```ruby
+#/app/controllers/spree/prescirptions_controller.rb
+module Spree
+  class PrescriptionsController < Spree::StoreController
+    def index
+      # Controller logic
+    end
+  end
+end
+```
+
+To add an entire part to spree it should be interesting to add a local controller and each new controller class will inherit of it, this class mother should inherit of Spree::StoreController. For example in the case above we will have:
+
+```ruby
+  class PrescriptionsController < LocalSpreeController
+```
+
+For the routing spree mount **Spree::Core::Engine** to `‘/’`, so every new route in relation with spree gem should be mount here also. In our example we will have:
+
+```ruby
+#/config/routes.rb
+Rails.application.routes.draw do
+  mount Spree::Core::Engine, at: '/'
+
+  Spree::Core::Engine.add_routes do
+    resources :prescriptions, only: [:index]
+  end
+end
+```
+
+The rest is equal.
+
+## Frontend tricks
+
+For the frontend there is 3 ways to modify it:
+
+1. You can use bootstrap, example [here](https://github.com/spree/spree/tree/master/frontend) and [here](https://github.com/twbs/bootstrap-sass/blob/master/assets/stylesheets/bootstrap/_variables.scss)
+
+2. The second option is to use deface, which permit to modify native view from spree, this solution is not a good way if the e-commerce need specific views with a design different from spree. At this [link](https://guides.spreecommerce.org/developer/deface_overrides_tutorial.html) you have a complete tutorial to learn more about it.
+
+3. The third option is - modify the HTML and add your CSS - by importing the view into your app and spree will use this views and not the native ones. For that spree *v3.4* allow to import all frontend in one command. This is a good solution for a simple design that do not need javascript action for animation or action and a flow that respect general spree flow. It is this solucion we used for Enjoy Vue and we will developed it.
+
+4. The fourth solution to used only the API and backoffice of Spree and use a frontend framework for views. This solution was never try but seem a good option for design far away from spree general structure. [Some developer ever done it](https://stackoverflow.com/questions/31332651/how-do-i-uninstall-spree-frontend).
```

Comment:
The fourth solution to be used is to use only the API and the backoffice of Spree and use a frontend framework for views. This solution was never tried but seems a good option for complex designs far away from the spree general structure. [Some developer ever done it](https://stackoverflow.com/questions/31332651/how-do-i-uninstall-spree-frontend).